.github/workflows/ci.yml

```diff
@@ -186,12 +191,43 @@ jobs:
           node-version: 16
       - run: npm install -g yarn
 
-      - name: Run integration test
+      - name: Prepare for integration tests
         run: |
           pushd integration-tests/
 
           # wait for creditcoin-node to initialize
           ./yarn-install-and-wait-for-creditcoin.sh
+          git checkout yarn.lock
+
+          popd
+
+      - name: Auto-update typedefs
+        id: update_typedefs
+        run: |
+          pushd creditcoin-js/
+
+          yarn build:types
+          yarn format
+
+          if [ -n "$(git status --short)" ]; then
+              git diff
+              echo "::set-output name=git_diff::true"
+          fi
+
+          popd
+
+      - name: Commit changes for typedefs
+        if: ${{ steps.update_typedefs.outputs.git_diff == 'true' }}
+        uses: EndBug/add-and-commit@v9
+        with:
+          author_name: gluwa-bot
+          author_email: creditcoin@gluwa.com
+          message: 'Auto-update creditcoin-js type definitions'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Execute integration tests
+        run: |
+          pushd integration-tests/
```

Comment:
I wonder if we should run the integration tests first and only commit the changes on success, so way we don't push a commit that breaks the tests (instead if they fail, we could just fail this CI job and emit a message saying typedefs need to be manually updated or something)

Reply:
I've pushed a commit which changes the behavior to commit the changes only if previous steps were successfull and there are changes to be committed. Also run a `git diff` in case integration tests fail.